Frameworks/Foundation/NSIndexPath.mm

```diff
@@ -62,16 +87,13 @@ - (instancetype)initWithIndex:(NSUInteger)index {
  @Status Interoperable
 */
 - (instancetype)initWithIndexes:(const NSUInteger*)indexes length:(NSUInteger)length {
-    _length = length;
-    _indexes = (unsigned*)IwMalloc(length * sizeof(unsigned));
-    if (_indexes == nil) {
-        [self release];
-        return nil;
+    if (self = [super init]) {
+        if (length != 0 && indexes != nullptr) {
+            _indexes.reserve(length);
+            _indexes.insert(_indexes.begin(), indexes, indexes + length);
+        }
```

Comment:
You have it such that, if you pass in an invalid index, you get a object rather nil.
I don't think that's what you want